### PR TITLE
Populate network drop-down in load balancer create/edit

### PR DESCRIPTION
**Pull request by jcwest**
Populate network drop-down in load balancer create/edit with values provided by back-end
Resolves issue #965.

### CODE REVIEW

File: app/scripts/modules/openstack/common/selectField.directive.js [hunk 1]
```diff
@@ -48,8 +48,12 @@ module.exports = angular.module('spinnaker.openstack.selectField.directive', [
...
             //if the selection changed as a result of updating the options
             if (scope.model !== previousSelection) {
-              //ensure that the change handler is invoked
-              scope.onChange();
+              //ensure that the change handler is invoked... but give time for the two-way binding to propagate the value
+              if( scope.onChange ){
+                $timeout(function() {
+                  scope.onChange();
+                }, 0);
+              }
```
Comment (anotherchrisberry) [inline app/scripts/modules/openstack/common/selectField.directive.js]: You can simplify #L52-56 to:
```javascript
$timeout(scope.onChange);
```
Even if there's no `onChange` set on the scope, it won't throw an exception.

Comment (jcwest) [inline app/scripts/modules/openstack/common/selectField.directive.js]: Makes sense. I'll make the change.

File: app/scripts/modules/openstack/common/selectField.directive.js [hunk 2]
```diff
@@ -48,8 +48,10 @@ module.exports = angular.module('spinnaker.openstack.selectField.directive', [
 
             //if the selection changed as a result of updating the options
             if (scope.model !== previousSelection) {
-              //ensure that the change handler is invoked
-              scope.onChange();
+              //ensure that the change handler is invoked... but give time for the two-way binding to propagate the value
+              if( scope.onChange ){
+                $timeout(scope.onChange);
+              }
```
Comment (anotherchrisberry) [inline app/scripts/modules/openstack/common/selectField.directive.js]: Don't need the wrapping `if` statement - just 
```javascript 
$timeout(scope.onChange)
```

Comment (jcwest) [inline app/scripts/modules/openstack/common/selectField.directive.js]: Yes, that is possible but it is a little bit less efficient as it results in a noop function callback being queued and a deferred/promise to be allocated unnecessarily.

**Comment by anotherchrisberry** [comment]
LGTM - needs a rebase

**Comment by jcwest** [comment]
Rebase done. Should be good to merge.

**Pull request merged**